pkgs/by-name/td/tdlib/package.nix

```diff
@@ -73,6 +75,11 @@ stdenv.mkDerivation {
     cmake --build native-build -j $NIX_BUILD_CORES
   '';
 
+  cmakeFlags = [
+    (lib.cmakeBool "TD_E2E_ONLY" tde2eOnly)
+    (lib.cmakeBool "TD_INSTALL_STATIC_LIBRARIES" tde2eOnly)
```

Comment:
I believe that's not what you really want as you basically disable anything but tdjson (i.e. all the native C++ tdlib API gets disabled)

Reply:
Ah, it should rather not pass those arguments at all if not `tde2eOnly`, instead of turning them off explicitly? So more like:
```nix
cmakeFlags = lib.optionals tde2eOnly [
  "-DTD_E2E_ONLY=ON"
  "-DTD_INSTALL_STATIC_LIBRARIES=ON"
];
```